sktime/transformations/series/compose.py

```diff
@@ -430,3 +433,145 @@ def _check_is_pdseries(z):
         z = z.to_frame()
         is_series = True
     return z, is_series
+
+
+class FitInTransform(BaseTransformer):
+    """Transformer compostion to always fit a given transformer on the transform data only.
+
+    Some transformers like Imputer(method="mean") might be more accurate for some use
+    cases if they are fitted only on the transform/predict data. The FitInTransform
+    transformer can be best used in a pipeline context where train data is different to
+    the transform/predict data.
+
+    Parameters
+    ----------
+    transformer : Estimator
+        scikit-learn-like or sktime-like transformer to fit and apply to series.
+
+    Attributes
+    ----------
+    transformer_ : Fitted transformer.
+
+    Examples
+    --------
+    >>> from sktime.datasets import load_longley
+    >>> from sktime.forecasting.naive import NaiveForecaster
+    >>> from sktime.forecasting.base import ForecastingHorizon
+    >>> from sktime.forecasting.compose import ForecastingPipeline
+    >>> from sktime.forecasting.model_selection import temporal_train_test_split
+    >>> from sktime.transformations.series.compose import FitInTransform
+    >>> from sktime.transformations.series.impute import Imputer
+
+    >>> y, X = load_longley()
+    >>> y_train, y_test, X_train, X_test = temporal_train_test_split(y, X)
+    >>> fh = ForecastingHorizon(y_test.index, is_relative=False)
+
+    >>> # we want to fit the Imputer only on the predict (=transform) data.
+    >>> # note that NaiveForecaster cant use X data, this is just a show case.
+    >>> pipe = ForecastingPipeline(
+    ...     steps=[
+    ...         ("imputer", FitInTransform(Imputer(method="mean"))),
+    ...         ("forecaster", NaiveForecaster()),
+    ...     ]
+    ... )
+    >>> pipe.fit(y_train, X_train)
+    ForecastingPipeline(...)
+    >>> y_pred = pipe.predict(fh=fh, X=X_test)
+    """
+
+    def __init__(self, transformer):
+        self.transformer = transformer
+        super(FitInTransform, self).__init__()
+        self.clone_tags(transformer, None)
+        self.set_tags(**{"fit_is_empty": True})
+        # self.set_tags(handles-missing-data=True)
+
+    def _transform(self, X, y=None):
+        """Transform X and return a transformed version.
+
+        private _transform containing core logic, called from transform
+
+        Parameters
+        ----------
+        X : Series or Panel of mtype X_inner_mtype
+            if X_inner_mtype is list, _transform must support all types in it
+            Data to be transformed
+        y : Series or Panel of mtype y_inner_mtype, default=None
+            Additional data, e.g., labels for transformation
+
+        Returns
+        -------
+        transformed version of X
+        """
+        self.transformer_ = clone(self.transformer)
+        self.transformer_.fit(X=X, y=y)
+        return self.transformer_.transform(X=X, y=y)
```

Comment:
this will fail the "transform does not change state" test.

The way to avoid this would be to not write to `self` at all, just remove the `transformer_` attribute, and create the clone locally in `_transform`.

Reply:
but I need to write to `self` in order to enable `inverse_transform` 🤔

Reply:
would `inverse_transform` not call `fit.inverse_transform`?

Reply:
what is `fit` here?

Reply:
of the inner trafo